src/web/EmployerFavouritesWeb/Infrastructure/AzureTableStorageFavouritesRepository.cs

```diff
@@ -0,0 +1,127 @@
+            {
+                var table = await GetTable();
+                TableOperation retrieveOperation = TableOperation.Retrieve<ApprenticeshipFavouritesEntity>(employerAccountId, ApprenticeshipFavouritesEntity.ROW_KEY);
+                
+                TableResult result = await _retryPolicy.ExecuteAsync(context => table.ExecuteAsync(retrieveOperation), new Context(nameof(GetApprenticeshipFavourites)));
+                ApprenticeshipFavouritesEntity entity = result.Result as ApprenticeshipFavouritesEntity;
+                
+                if (entity != null)
+                {
+                    _logger.LogTrace("\t{0}\t{1}\t{2}", entity.PartitionKey, entity.RowKey, JsonConvert.SerializeObject(entity.Favourites));
+                }
+
+                return entity?.ToApprenticeshipFavourites();
+            }
+            catch (StorageException ex)
+            {
+                _logger.LogError(ex, "Unable to Get Apprenticeship Favourites");
+                throw;
+            }
+        }
+
+        public async Task SaveApprenticeshipFavourites(string employerAccountId, ApprenticeshipFavourites apprenticeshipFavourite)
+        {
+            if (apprenticeshipFavourite == null)
+            {
+                throw new ArgumentNullException("entity");
+            }
+
+            try
+            {
+                var table = await GetTable();
+                var entity = new ApprenticeshipFavouritesEntity(employerAccountId, apprenticeshipFavourite);
+
+                // Create the InsertOrReplace table operation
+                TableOperation insertOrMergeOperation = TableOperation.InsertOrMerge(entity);
+
+                // Execute the operation.
+                TableResult result = await _retryPolicy.ExecuteAsync(context => table.ExecuteAsync(insertOrMergeOperation), new Context(nameof(SaveApprenticeshipFavourites)));
+                ApprenticeshipFavouritesEntity insertedCustomer = result.Result as ApprenticeshipFavouritesEntity;
+            }
+            catch (StorageException ex)
+            {
+                 _logger.LogError(ex, "Unable to Save Apprenticeship Favourites");
+                throw;
+            }
+        }
+
+        private async Task<CloudTable> GetTable()
+        {
+            CloudTableClient tableClient = _storageAccount.CreateCloudTableClient(new TableClientConfiguration());
+
+            CloudTable table = tableClient.GetTableReference(TABLE_NAME);
+                    
+            var createdTable = await _retryPolicy.ExecuteAsync(context => table.CreateIfNotExistsAsync(), new Context(nameof(GetTable)));
+            
+            if (createdTable)
+            {
+                _logger.LogDebug($"Created Table named: {TABLE_NAME}");
+            }
+            else
+            {
+                _logger.LogDebug($"Table {TABLE_NAME} already exists");
+            }
+
+            return table;
+        }
+
+        private Polly.Retry.AsyncRetryPolicy GetRetryPolicy()
```

Comment:
Should the retry policy be injected using the interface and have the calling application define the retry policy rather than have it explicitly set? this way the retry policy would only ever be defined once in an application? maybe a little redundant currently where its all in same project!

Reply:
I would say in this particular scenario probably not. I think retry functionality can be quite tied to the implementing technology. e.g. CosmosDb returns a time to wait when it's too busy/ request rate too large, also very often it's built into the client library itself.